core/src/main/java/org/infinispan/configuration/cache/Configurations.java

```diff
@@ -95,6 +95,10 @@ public static boolean isClustered(GlobalConfiguration globalConfiguration) {
       return globalConfiguration.transport().transport() != null;
    }
 
+   public static boolean isStateTransferStore(StoreConfiguration storeConfiguration) {
+      return !storeConfiguration.shared() && storeConfiguration.fetchPersistentState();
```

Comment:
I wonder if instead we should have a validation check of cache store config to not allow `fetchPersistentState` and `shared` together as they don't make any sense and throw an exception. Then this just becomes a simple check of `fetchPersistentState`.

Reply:
Sounds good. I didn't think that would work because `fetchPersistentState` is enabled by default and we want to allow users enabling `shared` without also disabling `fetchPersistentState` explicitly, but I was wrong, `fetchPersistentState` is disabled by default.